modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/DataSetDisplay.js

```diff
@@ -159,57 +120,81 @@ function DataSetDisplay({
 			delta,
 			pageNumber,
 			sorting
-		)
-			.then(updateDataSetItems)
-			.then(() => {
-				const {message, showSuccessNotification} = successNotification;
-
-				if (showSuccessNotification) {
-					openToast({
-						message:
-							message ||
-							Liferay.Language.get('table-data-updated'),
-						type: 'success',
-					});
-				}
+		).catch((error) => {
+			logError(error);
+			openToast({
+				message: Liferay.Language.get('unexpected-error'),
+				type: 'danger',
+			});
 
-				setLoading(false);
-				Liferay.fire(DATASET_DISPLAY_UPDATED, {id});
-			})
-			.catch((error) => {
-				logError(error);
-				setLoading(false);
+			throw error;
+		});
+	}, [apiURL, currentURL, delta, filters, pageNumber, searchParam, sorting]);
+
+	const requestComponent = useCallback(() => {
+		if (
+			!CurrentViewComponent &&
+			(contentRendererModuleURL || contentRenderer)
+		) {
+			return (contentRenderer
+				? getViewContentRenderer(contentRenderer)
+				: getJsModule(contentRendererModuleURL)
+			).catch((error) => {
+				logError(
+					`Requested module: ${contentRendererModuleURL} not available`,
+					error
+				);
 				openToast({
 					message: Liferay.Language.get('unexpected-error'),
 					type: 'danger',
 				});
+
+				throw error;
 			});
+		}
+
+		return Promise.resolve(CurrentViewComponent);
+	}, [contentRenderer, contentRendererModuleURL, CurrentViewComponent]);
+
+	const isMounted = useIsMounted();
+
+	function updateDataSetItems(dataSetData) {
+		setTotal(dataSetData.totalCount);
+		updateItems(dataSetData.items);
 	}
 
-	const {apiURL} = useContext(AppContext);
 	useEffect(() => {
-		if (apiURL) {
-			getData(
-				apiURL,
-				currentURL,
-				filters.filter((filter) => filter.value),
-				searchParam,
-				delta,
-				pageNumber,
-				sorting,
-				false
-			);
-		}
-		// eslint-disable-next-line react-hooks/exhaustive-deps
+		let resolved = false;
+
+		Promise.race([
+			delay(200).then(() => {
+				if (isMounted && !resolved) {
```

Comment:
This needs to be `isMounted()`, because it is _always_ truthy. You have to call it to get the actual value out of it.

Common mistake to make, which is why we plan to make [a lint rule for it](https://github.com/liferay/eslint-config-liferay/issues/157). I am way behind on that stuff because the pandemic has basically soaked up a lot of bandwidth, but I think it's time to actually prioritize it soon.

Same issue on line 177 below.

Reply:
Ah, right. I wasn't wearing my glasses when I wrote this.

![](https://media2.giphy.com/media/njYiRJVemOny8bHJNt/giphy.gif?cid=5a38a5a2c3anq9nw99diqdvvt642ihg351farwdd7s0v2gom&rid=giphy.gif)